tools/lldb/dylan/summaries/__init__.py

```diff
@@ -48,23 +48,29 @@ def dylan_object_summary(value, internal_dict):
     if summary is not None:
       return '{%s: %s}' % (class_name, summary)
     else:
-      return '{%s}' % class_name
+      return '{%s}' % (class_name,)
   else:
     try:
-      return '{%s}' % dylan_object_class_name(value)
+      return '{%s}' % (dylan_object_class_name(value),)
     except:
       return '{uninitialized}'
 
 def dylan_byte_character_summary(value, internal_dict):
   byte_character = dylan_byte_character_value(value)
-  return '{<byte-character>: %s (%s)}' % (byte_character, ord(byte_character))
+  if byte_character.isprintable():
+    return '{<byte-character>: %s (0x%x)}' % (byte_character, ord(byte_character))
+  else:
+    return '{<byte-character>: (0x%x)}' % (ord(byte_character),)
```

Comment:
Since I might be responsible for this I want to expand on my previous comment about string interpolation. :-)  Using `"%s" % foo` instead of `"%s" % (foo,)` is totally fine if you're sure the value of `foo` isn't a tuple. So if you call `ord` you don't need the tuple wrapper.

(Also it's fine to keep the tuple wrapper! And apologies if you already know all this.)

(Also also, most of my Python knowledge predates Python3.)

Reply:
I did know, then forgot, then misremembered after your comment. I thought the singleton tuples were the correct way and the tuple-less way was a syntax 'concession' which was frowned on. I've re-read the docs now!